data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/PluginDurationDeserializer.java

```diff
@@ -0,0 +1,69 @@
+package com.amazon.dataprepper.plugin;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This deserializer is used for all plugin configurations that use a {@link Duration} type that is mapped to by Jackson in the {@link PluginConfigurationConverter}.
+ * It supports ISO 8601 notation ("PT20.345S", "PT15M", etc.) and simple durations for
+ * seconds (60s) and milliseconds (100ms). It does not support combining the units for simple durations ("60s 100ms" is not allowed).
+ * Whitespace is ignored and leading zeroes are not allowed.
+ * @since 1.3
+ */
+public class PluginDurationDeserializer extends StdDeserializer<Duration> {
```

Comment:
Please make this package protected. If needed further, we can make it public in the future.